examples/c/minimal_legacy.bpf.c

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2020 Facebook */
+#define BPF_NO_GLOBAL_DATA
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+// Create an array with 1 entry instead of a global variable
+// which does not work with older kernels
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, u32);
+	__type(value, pid_t);
+} my_pid SEC(".maps");
+
+SEC("tp/syscalls/sys_enter_write")
+int handle_tp(void *ctx)
+{
+	u32 index = 0;
+	pid_t pid = bpf_get_current_pid_tgid() >> 32;
+	pid_t *user_pid = bpf_map_lookup_elem(&my_pid, &index);
+	if (user_pid)
+	{
+		if (*user_pid == pid)
```

Comment:
combine two ifs and invert the condition to exit early?

```
if (user_pid && *user_pid == pid)
    return 1;

bpf_printk(...);
return 0;
```

Reply:
Shouldn't I still return 0? and exit early when the pid does not match:

if (user_pid && *user_pid != pid) return 0;

It may be better just to say:

if (user_pid && *user_pid == pid)
   bpf_printk(...);

Reply:
so 0 vs 1 in tracepoints is not about returning error, it's whether perf subsystem will generate event for tracepoint or not. Returning 0 prevents it, which can interfere with some perf uses. So returning 1 always is a better idea in general. It's just a little known detail about BPF tracepoint programs, and it usually doesn't cause problems (but sometimes it does, unfortunately).

As for exiting early, it's just cleaner because it leaves the rest of logic more linear. Anyone using this example to extend it and play with it will have easier time extending it if you exit early.